entproto/internal/entprototest/service_test.go

```diff
@@ -14,30 +14,140 @@
 
 package entprototest
 
+import (
+	"fmt"
+)
+
+type method struct {
+	name         string
+	input        string
+	output       string
+	formatInput  bool
+	formatOutput bool
+}
+
+func (m *method) getFormatInput(name string) string {
+	return fmt.Sprintf(m.input, name)
+}
+func (m *method) getFormatOutput(name string) string {
+	return fmt.Sprintf(m.output, name)
+}
+
+var (
+	methodCreate = method{
+		name:         "Create",
+		input:        "Create%sRequest",
+		output:       "%s",
+		formatInput:  true,
+		formatOutput: true,
+	}
+
+	methodGet = method{
+		name:         "Get",
+		input:        "Get%sRequest",
+		output:       "%s",
+		formatInput:  true,
+		formatOutput: true,
+	}
+
+	methodUpdate = method{
+		name:         "Update",
+		input:        "Update%sRequest",
+		output:       "%s",
+		formatInput:  true,
+		formatOutput: true,
+	}
+
+	methodDelete = method{
+		name:         "Delete",
+		input:        "Delete%sRequest",
+		output:       "Empty",
+		formatInput:  true,
+		formatOutput: false,
+	}
+)
+
 func (suite *AdapterTestSuite) TestServiceGeneration() {
-	fd, err := suite.adapter.GetFileDescriptor("BlogPost")
-	suite.Require().NoError(err)
```

Comment:
This test code gets a little too clever for what it needs to do.  I had to read it a few times to understand why you need to formatInput/formatOutput, etc. 

That's just too much logic. Tests should be self explanatory, very _very_ simple (since they aren't tested themselves). DRY is not the most important design principle. Copy-paste is perfectly fine.

I wrote about this topic in my personal blog some time ago:
https://rotemtam.com/2020/05/18/the-dry-principle-is-bad-advice/

I would go back to just having short assertions about what you're verifying.